frontend/tests/unit/stores/settings.spec.ts

```diff
@@ -0,0 +1,30 @@
+import '~/utils/mocks/globals';
+
+import { createPinia, setActivePinia } from 'pinia';
+
+import { useSettings } from '@/stores/settings';
+import { WalletType } from '@/types/settings';
+
+describe('settings store', () => {
+  beforeEach(() => {
+    const pinia = createPinia();
+    setActivePinia(pinia);
+  });
+
+  describe('setConnectedWallet', () => {
+    it('can set a connected wallet', () => {
+      const settings = useSettings();
+      settings.setConnectedWallet(WalletType.Metamask);
+      expect(settings.$state.settings?.connectedWallet).toBe(WalletType.Metamask);
+    });
+  });
+
+  describe('connectedWallet getter', () => {
+    it('can get connected wallet value', () => {
+      const settings = useSettings();
+      settings.setConnectedWallet(WalletType.Metamask);
```

Comment:
Please try to avoid using other functions that are under test within the test of another one. This means if one is broken also the other test fails. Rather set the expected state manually via `settings.settings.connectedWallet = WalletType.MetaMask;`. Following this concept doesn't always work, but here it does and should be followed.

Reply:
That was a stupid assumption as well.. It seems I was asleep while implementing this because I assumed it will not reach this test case except passing the previous one!